Python/Product/PythonTools/PythonTools/Repl/PythonDebugProcessReplEvaluator.cs

```diff
@@ -275,18 +276,54 @@ public PythonDebugProcessReplEvaluator(IServiceProvider serviceProvider, PythonP
         }
 
         public override CompletionResult[] GetMemberNames(string text) {
-            if (string.IsNullOrEmpty(text) && _currentFrameLocals != null) {
+            if (_currentScopeName == CurrentFrameScopeFixedName && string.IsNullOrEmpty(text) && _currentFrameLocals != null) {
                 return _currentFrameLocals.ToArray();
             };
-            // TODO: Implement child members
-            return Array.Empty<CompletionResult>();
+
+            return _serviceProvider.GetUIThread().InvokeTaskSync(async () => await GetMemberNamesAsync(text), CancellationToken.None);
         }
 
         public override OverloadDoc[] GetSignatureDocumentation(string text) {
             // TODO: implement this
             return new OverloadDoc[0];
         }
 
+        private async Task<CompletionResult[]> GetMemberNamesAsync(string text) {
+            // TODO: implement support for getting members for module scope,
+            // not just for current frame scope
+            if (_currentScopeName == CurrentFrameScopeFixedName) {
+                var frame = GetFrames().SingleOrDefault(f => f.FrameId == _frameId);
+                if (frame != null) {
+                    AutoResetEvent completion = new AutoResetEvent(false);
+                    PythonEvaluationResult result = null;
+
+                    var expression = string.Format(CultureInfo.InvariantCulture, "str(dir({0}))", text ?? "");
```

Comment:
Maybe do `':'.join(dir({0}))` as the expression and then split on `:`?

Reply:
Also, do we need to take into account not evaluating calls? I think that's handled already, but I don't recall exactly...